feat(launchpad): stream wallet events to front-end

alex1307 opened the pull request:
Description

Provide a new Tari event stream for:
	- transaction received (source: wallet)
	- transaction sent (source: wallet)
	- transaction cancelled (source: wallet)
	- transaction mined but unconfirmed (source: wallet)
	- transaction mined and confirmed (source: wallet)

Expose a streaming interface for all data we log using.
Read all events from database and stream them over grpc streaming endpoint.


Motivation and Context
The launchpad users want to see the actual mining progress and all mining relalted events on the UI.
Hence we should forward the data/log streams from the underlying apps to the front-end.


How Has This Been Tested?

!!! TO_DO: The change IS NOT TESTED.
The change should be tested manually along with base node, tor, and launchpad.

tarnas14 commented:
@alex1307 could you make sure that this PR should not point to `155_stream_wallet_events_to_front_end` instead of `launchpad_such_wow`, I think it should, first commits look similar

~also, there are unverified commits, you should rectify it before we can merge it~
nvm they're signed but email is different so github complains

CjS77 commented on applications/tari_console_wallet/README.md:
```diff
@@ -263,3 +263,5 @@ Run a series of commands from a given script. The commands should be formatted t
 ## Recovery mode
 
 todo docs
+
+#
```
?

alex1307 replied:
I will remove it. It was a test for my first verified commit.

CjS77 commented on applications/tari_console_wallet/src/grpc/mod.rs:
```diff
@@ -3,4 +3,54 @@
 pub use self::wallet_grpc_server::*;
+
+
+pub enum TransactionWrapper {
```
Now that we're not going via Notifier, this bridging type is unnecessary. But I'm ok with doing this in a follow-up PR.

CjS77 commented on applications/tari_console_wallet/src/main.rs:
```diff
@@ -50,7 +51,18 @@ use tari_shutdown::Shutdown;
+    recovery::get_seed_from_seed_words,
+};
+
+// lazy_static! {
+//     static ref WALLET_EVENT_LISTENER: WalletEventListener = {
+//         let (sender, receiver) = crossbeam::channel::unbounded::<WalletEventMessage>();
+//         WalletEventListener::new(sender, receiver)
+//     };
+// }
```
delete

CjS77 commented on applications/tari_console_wallet/src/main.rs:
```diff
@@ -65,7 +77,19 @@ mod ui;
 mod wallet_modes;
 
-/// Application entry point
+// /// Application entry point
+//
+// pub struct WalletEventListener {
+//     sender: Sender<WalletEventMessage>,
+//     receiver: Receiver<WalletEventMessage>,
+// }
+//
+// impl WalletEventListener {
+//     pub fn new(sender: Sender<WalletEventMessage>, receiver: Receiver<WalletEventMessage>) -> Self {
+//         WalletEventListener { sender, receiver }
+//     }
+// }
```
delete

CjS77 reviewed (commented):
You can test this locally by pointing a gRPC client, like Postman at the wallet and seeing that the events come in

CjS77 reviewed (dismissed):
Let's merge as is to allow FE development to continue apace.

I've marked some comments as to where to improve things down the line.

alex1307 committed:
feat(launchpad): stream wallet events to front-end
Description

Provide a new Tari event stream for:
	- transaction received (source: wallet)
	- transaction sent (source: wallet)
	- transaction cancelled (source: wallet)
	- transaction mined but unconfirmed (source: wallet)
	- transaction mined and confirmed (source: wallet)

Expose a streaming interface for all data we log using.
Read all events from database and stream them over grpc streaming endpoint.

Motivation and Context
The launchpad users want to see the actual mining progress and all mining relalted events on the UI.
Hence we should forward the data/log streams from the underlying apps to the front-end.

How Has This Been Tested?

!!! TO_DO: The change IS NOT TESTED.
The change should be tested manually along with base node, tor, and launchpad.

CjS77 reviewed (approved):
Merging as is
